File: OOP/src/com/msb/oop/polymorphism/Cat.java
package com.msb.oop.polymorphism;

/**
 * @Auther: Huanyu Lee
 * @Date: 2022/1/22  12:24
 * @Description: 多态
 * @version: 1.0
 */
public class Cat extends Animal{
    protected double weight;
    public Cat() {
    }
    public Cat(String name, int age, double weight) {
        super(name, age);
        this.weight = weight;
    }

    @Override
    public void shout(){
        System.out.println("我是小猫，我可以喵喵叫……");
    }
    @Override
    public void eat(){
        System.out.println("我是小猫，我喜欢吃鱼……");
    }
    public void scratch(){
        System.out.println("我是小猫，我生气了会挠人……");
    }

    public double getWeight() {
        return weight;
    }
    public void setWeight(double weight) {
        this.weight = weight;
    }
}

File: src/ch10_oo/inheritance/item/cat.go
// TODO: golang中的继承是真正意义上的继承吗？
// 严格意义来说，golang中没有继承，它是一个“语法糖”

package item

import (
	"fmt"
)

type Cat struct {
	// 加入匿名结构体
	Animal
}

func (c *Cat) Scratch() {
	fmt.Println(c.GetName(), ": 会挠人...")
}

//func (c *Cat) String() string {
//	return fmt.Sprintf("Cat{name = %v, age = %v}", c.Name, c.Age)
//}
